common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -146,6 +178,44 @@ func (vi *valuesInterceptor) Values(name string, values ...interface{}) ([]inter
 		default:
 		}
 
+		switch fieldType {
+		case enumspb.INDEXED_VALUE_TYPE_INT, enumspb.INDEXED_VALUE_TYPE_DOUBLE:
+			switch v := value.(type) {
+			case int64, float64:
+			// nothing to do
+			case string:
+				// ES can do implicit casting if the value is numeric
+				if _, err := strconv.ParseFloat(v, 64); err != nil {
+					return nil, query.NewConverterError(
+						"invalid value for search attribute %s of type %s: %#v", name, fieldType.String(), value)
+				}
+			default:
+				return nil, query.NewConverterError(
+					"invalid value for search attribute %s of type %s: %#v", name, fieldType.String(), value)
+			}
+		case enumspb.INDEXED_VALUE_TYPE_BOOL:
+			switch value.(type) {
+			case bool:
+			// nothing to do
+			default:
+				return nil, query.NewConverterError(
+					"invalid value for search attribute %s of type %s: %#v", name, fieldType.String(), value)
+			}
+		case enumspb.INDEXED_VALUE_TYPE_DATETIME:
+			switch v := value.(type) {
+			case int64:
+				value = time.Unix(0, v).UTC().Format(time.RFC3339Nano)
```

Comment:
is this (treat int as nanoseconds) current behavior?

Reply:
Yes, this is the default behavior